File: solutions/javascript/atbash-cipher/1/atbash-cipher.js
/**
 * The alphabet used for encoding and decoding.
 * @constant {string}
 */
const ALPHABET = 'abcdefghijklmnopqrstuvwxyz';

/**
 * Encodes a given message using the Atbash cipher.
 * Removes spaces, periods, and commas, and converts to lowercase.
 * Groups the encoded message into chunks of five characters.
 * 
 * @param {string} message - The message to encode.
 * @returns {string} The encoded message.
 */
export const encode = (message) => {
  // Remove spaces, periods, and commas, and convert to lowercase
  const sanitizedMessage = message.toLowerCase().replace(/[ .,]/g, '');
  
  const encodedMessage = [...sanitizedMessage]
    .map((char) => {
      if (ALPHABET.includes(char)) {
        return ALPHABET[ALPHABET.length - 1 - ALPHABET.indexOf(char)];
      }
      return char;
    })
    .join('');
  
  // Group the encoded message into chunks of five characters
  return encodedMessage.match(/.{1,5}/g).join(' ');
};

/**
 * Decodes a given message using the Atbash cipher.
 * Removes spaces from the encoded message.
 * 
 * @param {string} message - The message to decode.
 * @returns {string} The decoded message.
 */
export const decode = (message) => {
  return encode(message).replace(/ /g, '');
};

File: solutions/python/atbash-cipher/2/atbash_cipher.py
from string import ascii_letters

ATBASH_CIPHER = str.maketrans(ascii_letters, ascii_letters.lower()[::-1])


def encode(plain_text):
    """
    Encodes the given plain text using the Atbash cipher.

    The Atbash cipher is a monoalphabetic substitution cipher where each letter of
    the alphabet is replaced by its reverse counterpart. This function first decodes
    the plain text using the Atbash cipher logic and then formats the encoded text into
    groups of five characters separated by spaces.

    Args:
        plain_text (str): The text to be encoded.

    Returns:
        str: The encoded text formatted in groups of five characters.
    """
    encoded = decode(plain_text)
    return " ".join(
        encoded[index : index + 5] for index in range(0, len(encoded), 5)
    )


def decode(ciphered_text):
    """
    Decodes the given ciphered text using the Atbash cipher.

    This function removes any non-alphanumeric characters from the input and then
    translates the remaining characters using the Atbash cipher mapping.

    Args:
        ciphered_text (str): The text to be decoded.

    Returns:
        str: The decoded text after applying the Atbash cipher.
    """
    return "".join(c for c in ciphered_text if c.isalnum()).translate(
        ATBASH_CIPHER
    )
